src/ploomber/cloud/api.py

```diff
@@ -170,12 +171,25 @@ def run_detail_print(run_id):
     tasks = out['tasks']
     run = out['run']
 
+    # pipeline has four status
     if run['status'] == 'created':
         click.echo('Run created...')
-    elif tasks:
+    elif run['status'] == 'aborted':
+        # both failed and aborted pipeline contain tasks
+        click.echo('Pipeline aborted...')
+        click.echo(Table.from_dicts(tasks))
+    elif run['status'] == 'failed':
+        click.echo('Pipeline failed...')
         click.echo(Table.from_dicts(tasks))
     else:
-        click.echo('Pipeline up-to-date, no tasks scheduled for this run.')
+        click.echo('Pipeline finished...')
+        if tasks:
+            # normal finish, print tasks
+            click.echo(Table.from_dicts(tasks))
+        else:
+            # pipeline does not run at all
+            click.echo('Pipeline finished due to no newly triggered tasks,'
+                       ' try build with --force')
```

Comment:
on 192, try running: `ploomber cloud build --force`.

Also, what's separating it from a regular finish? like if there's a pipeline that finished as expected, I believe it'll fall into the else clause right?

Reply:
I thought there are two situation for finishing? 1. regular finish with tasks, we print out task information on line 188. 2. finish because of no new tasks run, there will be no tasks, so we go to line 191 print the message?

Reply:
Yes you're right, the regular finish will print the tasks and a message of pipeline finished in this case.